reactivesocket-client/src/main/java/io/reactivesocket/client/LoadBalancer.java

```diff
@@ -468,22 +471,44 @@ public synchronized String toString() {
 
     @Override
     public Publisher<Void> close() {
-        return s -> {
-            Publishers.afterTerminate(onClose(), () -> {
-                synchronized (this) {
-                    factoryRefresher.close();
-                    activeFactories.clear();
-                    activeSockets.forEach(rs -> {
-                        try {
-                            rs.close();
-                        } catch (Exception e) {
-                            logger.warn("Exception while closing a ReactiveSocket", e);
+        return subscriber -> {
+            subscriber.onSubscribe(EmptySubscription.INSTANCE);
+
+            synchronized (this) {
+                factoryRefresher.close();
+                activeFactories.clear();
+                AtomicInteger n = new AtomicInteger(activeSockets.size());
+
+                activeSockets.forEach(rs -> {
+                    rs.close().subscribe(new Subscriber<Void>() {
+                        @Override
+                        public void onSubscribe(Subscription s) {
+                            s.request(Long.MAX_VALUE);
+                        }
+
+                        @Override
+                        public void onNext(Void aVoid) {}
+
+                        @Override
+                        public void onError(Throwable t) {
+                            if (n.decrementAndGet() == 0) {
+                                subscriber.onComplete();
+                                closeSubject.subscribe(EmptySubscriber.INSTANCE);
+                                closeSubject.onComplete();
+                            }
+                        }
+
+                        @Override
+                        public void onComplete() {
+                            if (n.decrementAndGet() == 0) {
```

Comment:
you're doing the same thing in the onComplete and the onError - why not just call onComplete in the onError method?